pkg/webui/lib/components/with-auth.js

```diff
@@ -13,57 +13,124 @@
-import { connect } from 'react-redux'
+import { withRouter } from 'react-router-dom'
+import Query from 'query-string'
+import { defineMessages } from 'react-intl'
+
+import sharedMessages from '../shared-messages'
+import { selectApplicationRootPath } from '../../lib/selectors/env'
+import { createFrontendError } from '../../lib/errors/utils'
 import PropTypes from '../prop-types'
+import Spinner from '../../components/spinner'
+import Message from './message'
+
+// Define a minimum set of rights, without which it makes no sense to access the
+// console
+const minimumRights = ['RIGHT_APPLICATION', 'RIGHT_GATEWAY', 'RIGHT_ORGANIZATION']
 
-import { withEnv } from '../../lib/components/env'
+const m = defineMessages({
+  errTooFewRights: 'Your account does not possess sufficient rights to use the console.',
+  errTooFewRightsTitle: 'Insufficient rights',
+  errStateRequested:
+    'Your account still needs to be approved by an administrator. You will receive a confirmation email once your account is approved.',
+  errStateRequestedTitle: 'Account unapproved',
+  errStateRejected: 'Your account has been rejected by an administrator.',
+  errStateRejectedTitle: 'Account rejected',
+  errStateSuspended:
+    'Your account has been suspended by an administrator. Please contact support for further information about your account status.',
+  errStateSuspendedTitle: 'Account suspended',
+  errEmailValidation: 'Your account is restricted until your email address has been validated.',
+  errEmailValidationTitle: 'Email validation pending',
+})
 
 /**
  * Auth is a component that wraps a tree that requires the user
  * to be authenticated.
  *
  * If no user is authenticated, it renders the login view.
  */
-@withEnv
 @withRouter
-@connect(state => ({
-  fetching: state.user.fetching,
-  user: state.user.user,
-}))
 class Auth extends React.PureComponent {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    error: PropTypes.error,
+    errorComponent: PropTypes.elementType.isRequired,
+    fetching: PropTypes.bool.isRequired,
+    isAdmin: PropTypes.bool,
+    location: PropTypes.location.isRequired,
+    rights: PropTypes.rights,
+    user: PropTypes.user,
+  }
+  static defaultProps = {
+    user: undefined,
+    error: undefined,
+    isAdmin: undefined,
+    rights: undefined,
+  }
+
   render() {
     const {
       user,
       fetching,
+      error,
+      errorComponent,
       children,
-      env: { appRoot },
+      location,
+      rights,
+      isAdmin,
     } = this.props
 
     if (fetching) {
-      return null
+      return (
+        <Spinner center>
+          <Message content={sharedMessages.fetching} />
+        </Spinner>
+      )
+    }
+
+    let err
+
+    if (error) {
+      err = error
+    } else if (
+      // Check whether the user has at least basic rights, without which it
+      // makes no sense to access the console
+      Boolean(user) &&
+      !isAdmin &&
```

Comment:
the admin account can be suspended/rejected as well. the only allowed operation for a suspended admin account is to list and delete other accounts. Do we want to allow that? 

Also, while checking this I also found out that you are not logged out of the console when you delete your self. Most probably this is related to https://github.com/TheThingsNetwork/lorawan-stack/pull/2122/files#r391668222

Also, see https://github.com/TheThingsNetwork/lorawan-stack/issues/2144